### PR TITLE
UPS-1791 Create moosend api client

### PR DESCRIPTION
**Pull request by azisuazusa**
## What does this PR do?
This is a moosend go client. Provide all client APIs that we need for migrating to moosend.

## Why are we doing this? Any context or related work?
https://kitabisa.atlassian.net/browse/UPS-1791

## Why don't we use the API wrapper provided by moosend?
- The wrapper code seems generated by swagger-code gen, as you can see here: https://github.com/moosend/api-wrappers-go/blob/7d1ac96d9d721a92191175e1348eb50e8df0c68d/api_client.go#L8
- I have a hard time when I read through the code
- We don't even able to use that as a dependency in our projects because its package name https://github.com/moosend/api-wrappers-go/blob/7d1ac96d9d721a92191175e1348eb50e8df0c68d/api_client.go#L11 it will give us an error
- Our requirements just need several APIs, so we don't need the whole things

### CODE REVIEW

File: mailing_list/mailing_list.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+		err = fmt.Errorf("[moosend-client] %d:%s", resp.StatusCode, response.Error)
+		return
+	}
+
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		err = fmt.Errorf("[moosend-client] %d:%s", resp.StatusCode, err.Error())
+		return
+	}
```
Comment (ardityawahyu) [inline mailing_list/mailing_list.go:49]: why duplicate the code unmarshal?

File: subscriber/subscriber.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+		err = fmt.Errorf("[moosend-client] %d:%s", resp.StatusCode, response.Error)
+		return
+	}
+
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		err = fmt.Errorf("[moosend-client] %d:%s", resp.StatusCode, err.Error())
+		return
+	}
```
Comment (ardityawahyu) [inline subscriber/subscriber.go:50]: idem dengan di atas

File: subscriber/subscriber.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
```
Comment (ardityawahyu) [inline subscriber/subscriber.go:81]: idem

File: subscriber/subscriber.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
```
Comment (ardityawahyu) [inline subscriber/subscriber.go:118]: idem

File: subscriber/subscriber.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
```
Comment (ardityawahyu) [inline subscriber/subscriber.go:155]: idem

**Review by ardityawahyu — COMMENTED** [review]
check comment yak, pertanyaannya berlaku buat sisanya juga

File: mailing_list/mailing_list.go (hunk 1 above)
Comment (azisuazusa) [reply mailing_list/mailing_list.go:49]: hoiya ya, baru sadar struct respon yang dipake sama, otw fixing mas

**Comment by azisuazusa** [comment]
done  ya mas @ardityawahyu

**Review by ardityawahyu — APPROVED** [review]

**Review by AhmadDzakiN — APPROVED** [review]

**Pull request merged**